devito/types/equation.py

```diff
@@ -242,8 +241,8 @@ def solve(eq, target, **kwargs):
     for e, t in zip(as_tuple(eq), as_tuple(target)):
         # Try first linear solver
         try:
-            cc = linear_coeffs(e._eval_at(t).evaluate, t)
-            sols.append(-cc[1]/cc[0])
+            sol, = sympy.linsolve([e._eval_at(t).evaluate], t)
```

Comment:
was `linear_coeffs` removed? for retrocompatibility you may try:

```
try:
    from sympy.solvers.solveset import linear_coeffs
except ImportError:
    ...
```

Reply:
No that's the issue, it's still there but it's been modified so it doesn't work anymore. Basically it use to check if `eq.has(target)` and now it only checks `target in eq.free_symbols` which fails because only the dimensions are in it, so I'm tempted to make a dummy type just for solve that adds `target` to its free symbols to bypass it because `linsolve` is not an option it's way too slow.